autoscheduler/scraper/banner_requests.py

```diff
@@ -0,0 +1,184 @@
+""" The Banner Requests class, which is what we'll use to retrieve all of the data/courses
+    from Banner(aka Compassxe). While department scraping is pretty straightforward,
+    course scraping will need to use be concurrent/use async+await in order to minimize
+    the runtime of it.
+
+    Course retrieval example:
+        var banner = BannerRequests("201931")
+
+        depts = ["CSCE", "MATH", "CHEM"]
+        requests = [banner.search(dept) for dept in depts]
+
+        data = []
+        async with asyncio.gather(*data) as resp:
+            data.append(resp)
+
+    Department retrieval example:
+        term = "201931"
+        var banner = BannerRequests(term)
+
+        amount = 100
+        depts = banner.get_departments(term, amount) # Amount is optional, defaults to 300

```

Comment:
1. BannerRequests constructor doesn't take term code as an argument (right now it's passed as an argument to `search()`), but I think it probably should be.
2. I think we should abstract away using `asyncio` (and maybe even making BannerRequests objects), we could make a top-level function `async_search(term, depts)` function to do this, and do the same with departments to avoid having to make a BannerRequests object.

Reply:
> 1. BannerRequests constructor doesn't take term code as an argument (right now it's passed as an argument to search()), but I think it probably should.

Yeah I agree. I was going back and forth about it, but if they want to change the search to another term they can just change it by doing `banner.term_code = newterm`

> 2. I think we should abstract away using asyncio (and maybe even making BannerRequests objects), we could make a top-level function async_search(term, depts) function to do this, and do the same with departments to avoid having to make a BannerRequests object.

Yeah honestly the Course Search example is incorrect, as with the `search` function all you have to do is.

```python
data = banner.search(depts, term)
```

Forgot to update it after I changed the designs, will now.

Reply:
> Yeah honestly the Course Search example is incorrect, as that's the whole point of the `search` function. This is what the example should look like:
> 
> ```python
> data = await banner.search(depts, term)
> ```
> 
> Forgot to update it after I changed the designs, will now.

Since we're going to have to use these functions inside of our scraping commands, this won't work as it is now. When we start doing scraping, this how we can run search:
```
#in scrape-courses.py (function to synchronously run async):
def run_async(function, args):
    loop = asyncio.get_event_loop()
    res = loop.run_until_complete(asyncio.wait([asyncio.ensure_future(function(args))]))
    return res
```
And in `handle(self, *args, **options)`, put the following to get the data:
`data = run_async(test, depts)`
This shouldn't be included in this pull request (I'll implement it when I do the `scrape-courses` outline), but I'm putting this here just as a note.

Reply:
> Since we're going to have to use these functions inside of our scraping commands, this won't work as it is now. When we start doing scraping, this how we can run search:

Why exactly won't this work? Is it b/c the actual `scrape_courses` will be synchronous?

What's the need for `tasks = [asyncio.ensure_future(function(args))]`? Why couldn't we just do `loop.run_until_complete(function(args))`? That's how I'm doing it in the auto scheduler on my repo and it seems to work fine.

Also since I completely forgot about `loop.run_until_complete()`, should I transition the rest of the tests(and remove aio_test_case) to use this instead, rather than calling `await`? I personally think the `await` syntax is cleaner in terms of the tests.